hare/broker.go

```diff
@@ -155,21 +112,22 @@ func (b *Broker) handleMessage(ctx context.Context, msg []byte) error {
 	logger.Debug("broker received hare message")
 
 	msgLayer := hareMsg.Layer
-	if !b.Synced(ctx, msgLayer) {
-		return errNotSynced
-	}
```

Comment:
I don't have a clear understanding yet of what it means to be synced, but when a consensus process registers with the broker it checks that it is synced before registering. So it seemed unnecessary to further check this inside the broker, and also seemed like it would be confusing to the consensus process if after registering, messages just stopped being relayed by the broker (which is what would happen if this synced check fails. Also it seems like (I'm guessing here though) if there was a necessity to be synced (i.e. some data needs to be retrieved that can only be retrieved by a synced process) then the consensus process is going to run into that necessity and at that point will understand that it is not synced and that it can't further process messages and maybe Unregister as a result. That seems like a better flow than the broker just silently discarding all the messages for a consensus process.

Reply:
the state of synced means that it has sync'ed all ATXs, ballots, and blocks since genesis
this check is important for a new node syncing from genesis. without this check, a new node will keep seeing errors around ATXs because it doesn't know the identity that send this gossip message.

specifically, a syncing node will not register for a consensus process, but will keep receiving hare gossip messages.

> That seems like a better than the broker just silently discarding all the messages for a consensus process.

this is only true for a node that is already sync'ed and online. if a node were to go from synced -> notSynced state, then the node will silently stopped processing message. but this state transition is detectable during debugging. alternatively, you can detect this (already registered but not sync'ed anymore) and terminate the consensus process with error like `losing synced state`. 

btw, 

https://github.com/spacemeshos/go-spacemesh/blob/6271e39ddafa202c8061fe5143911eb0a0dff419/cmd/node/node.go#L730

already ensures that it is in a synced state before receiving hare gossip message. but it doesn't guarantee the beacon syncedness, which is different by epoch. that's why it's not straight-foward to add there.

Reply:
I've re-instated this check for syncedness.